test/helpers/kubectl.go

```diff
@@ -1391,6 +1407,29 @@ func (kub *Kubectl) waitForNPods(checkStatus checkPodStatusFunc, namespace strin
 		&TimeoutConfig{Timeout: timeout})
 }
 
+func (kub *Kubectl) waitForSinglePod(checkStatus checkPodStatusFunc, namespace string, filter string, timeout time.Duration) error {
+	body := func() bool {
+		pod := v1.Pod{}
+		err := kub.GetPods(namespace, filter).Unmarshal(&pod)
```

Comment:
For reviewers: see commit message for why I added this helper function. Curious if there's a simpler way to unmarshal the output when we have a single pod.

Reply:
I think it would be simpler once we switch to using `k8s.io/client-go`, instead wrapping `kubectl` and parsing output.

Reply:
I think this machinery also predates `kubectl wait`, you might want to try using it.

Reply:
> I think this machinery also predates `kubectl wait`, you might want to try using it.

I've opened #13909 for that.